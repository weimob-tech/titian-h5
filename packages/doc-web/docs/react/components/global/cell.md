---
title: 单元格
sidebar_custom_props:
  suffix: Cell
hide_title: true
pagination_prev: null
pagination_next: null
side_iframe_path: "#/cell"
---

# 单元格 _Cell_

**单元格为列表中的单个展示项，可使用多个单元格可以组合成一个单元格组。**

import TabsLink from '@site/src/components/tabsLink';

<TabsLink id="ticell-api" />

## 安装使用

```typescript showLineNumbers
import { TiCell, TiCellGroup } from '@titian-design/mobile-react';
```

## 基本用法

#### 单独使用单元格

```typescript tsx showLineNumbers link="https://codesandbox.io/s/ti-cell-8yv5md?resolutionWidth=320&resolutionHeight=675&file=/src/example/cell.tsx"
const App: React.FC = () => {
  return (
    <>
      <section>
        <TiCell title="标题" />
        <TiCell title="标题" desc="居右详细内容文字" />
        <TiCell title="标题" label="标题下方附加信息" />
        <TiCell title="标题" desc="自定义左侧图标" icon="home" />
        <TiCell title="标题" desc="自定义图标颜色" icon="home" color="#ff0000" />
        <TiCell title="标题" desc="自定义右侧图标" rightIcon="arrow-right" />
        <TiCell title="标题" desc="使用默认右侧箭头" arrow />
        <TiCell title="标题" desc="禁用单元格" disabled />
        <TiCell title="标题" desc="开启点击反馈" clickable />
        <TiCell title="标题" label="附加信息" desc="居上对齐" alignItems="start" />
        <TiCell title="标题" label="附加信息" desc="居中对齐" alignItems="center" />
        <TiCell title="标题" label="附加信息" desc="居下对齐" alignItems="end" />
        <TiCell title="标题" desc="不展示下方分割线" divider={false} />
      <section>
    </>
  )
}
```

#### 使用 `slot` 修改单元格内容

```typescript tsx showLineNumbers  link="https://codesandbox.io/s/ti-cell-8yv5md?resolutionWidth=320&resolutionHeight=675&file=/src/example/cell.tsx"
const App: React.FC = () => {
  return (
    <>
      <TiCell title="使用slot自定义左侧图标">
        <TiIcon name="home" slot="icon" />
      </TiCell>
    
      <TiCell title="使用slot自定义右侧图标">
        <TiIcon name="home" slot="right-icon" />
      </TiCell>
    
      <TiCell>
        <div slot="title">使用 slot 修改 title</div>
      </TiCell>
    
      <TiCell title="使用slot自定义label">
        <div slot="label">标题下方附加信息</div>
      </TiCell>
    
      <TiCell title="使用slot自定义desc">
        <div slot="desc">居右详细内容文字</div>
      </TiCell>
    </>
  )
}
```

#### 多个单元格组合在一起

```typescript tsx showLineNumbers link="https://codesandbox.io/s/ti-cell-8yv5md?resolutionWidth=320&resolutionHeight=675&file=/src/example/cell.tsx"
const App: React.FC = () => {
  return (
    <>
      <TiCellGroup title="单元格组标题">
        <TiCell title="标题" desc="详细内容文字" />
        <TiCell title="标题" desc="详细内容文字" />
        <TiCell title="标题" desc="详细内容文字" />
      </TiCellGroup>
    
      <TiCellGroup title="单元格组标题" subTitle="单元格组副标题">
        <TiCell title="标题" desc="详细内容文字" />
        <TiCell title="标题" desc="详细内容文字" />
        <TiCell title="标题" desc="详细内容文字" />
      </TiCellGroup>
    
      <TiCellGroup title="单元格组模式为卡片" mode="card">
        <TiCell title="标题" desc="详细内容文字" />
        <TiCell title="标题" desc="详细内容文字" />
        <TiCell title="标题" desc="详细内容文字" />
      </TiCellGroup>
    </>
  )
}
```

#### 为单元格绑定点击事件

```typescript tsx showLineNumbers  link="https://codesandbox.io/s/ti-cell-8yv5md?resolutionWidth=320&resolutionHeight=675&file=/src/example/cell.tsx"
const App: React.FC = () => {
  const handleClick = useCallback((e: Event) => {
    console.log('单元格点击事件');
  }, [])
  
  return <TiCell title="标题" onClick={handleClick} />
}
```

import Tabs from '@theme/Tabs';
import TabItem from '@theme/TabItem';


#### 通过  [::part](https://developer.mozilla.org/en-US/docs/Web/CSS/::part) 修改组件样式

<Tabs>
<TabItem value="js" label="app.ts">

```typescript tsx showLineNumbers  link="https://codesandbox.io/s/ti-cell-8yv5md?resolutionWidth=320&resolutionHeight=675&file=/src/example/cell.tsx"
const App: React.FC = () => {
  return (
    <>
      <TiCell title="自定义 title 根节点样式" extClass="custom-cell-ext-class" />
    <>
  )
}
```

</TabItem>
<TabItem value="css" label="custom-cell.css">

```css showLineNumbers
ti-cell::part(custom-cell-ext-class) {
  background-color: red;
}
```

</TabItem>
</Tabs>

#### 通过 [css变量](https://developer.mozilla.org/en-US/docs/Web/CSS/Using_CSS_custom_properties) 修改组件样式

<Tabs>
<TabItem value="js" label="app.ts">

```typescript tsx showLineNumbers  link="https://codesandbox.io/s/ti-cell-8yv5md?resolutionWidth=320&resolutionHeight=675&file=/src/example/cell.tsx"
const App: React.FC = () => {
  return (
    <section className="custom-cell-ext-class">
      <TiCell title="自定义 title 根节点样式"  />
    </section>
  )
}
```

</TabItem>
<TabItem value="css" label="custom-cell.css">

```css showLineNumbers
.custom-cell-ext-class {
  --cell-bg-color: red;
}
```

</TabItem>
</Tabs>


## TiCell API

### 属性 **Properties**

| 名称          | 类型      | 是否必填 | 默认值     | 说明                                                         | 备注                                                              |
| ------------- | --------- | -------- | ---------- | ------------------------------------------------------------ | ----------------------------------------------------------------- |
| title         | `string`  | 否       | -          | 标题                                                         | - |
| label         | `string`  | 否       | -          | 标题下方附加信息                                             | -                                                                 |
| desc          | `string`  | 否       | -          | 副标题，右侧描述信息                                         | -                                                                 |
| subDesc       | `string`  | 否       | -          | 副标题下面附加信息                                           | -                                                                 |
| arrow         | `boolean` | 否       | `false`     | 是否使用默认右侧向右箭头                                         | -                                                                 |
| required      | `boolean` | 否       | `false`      | 是否显示表单必填星号                                         |           -                                                        |
| disabled      | `boolean` | 否       | `false`      | 是否禁用单元格                                               | -                                                                 |
| icon          | `string`  | 否       | -          | 自定义左侧图标                                               | -                                                                 |
| iconSize      | `string`  | 否       | -          | 自定义左侧图标尺寸                                           | -                                                                 |
| rightIcon     | `string`  | 否       | -          | 自定义右侧侧图标                                             | -                                                                 |
| rightIconSize | `string`  | 否       | -          | 自定义右侧图标尺寸                                           | -                                                                 |
| color         | `string`  | 否       | -          | 所有图标颜色，如果需要修改某个单一图标颜色，可使用自定义类名 | -                                                                 |
| clickable     | `boolean` | 否       | `false`       | 开启点击反馈                                                   | -                                                                 |
| alignItems    | `string`  | 否       | -          | 单元格的横向对其方式, 具体值可见 [align-items](https://developer.mozilla.org/en-US/docs/Web/CSS/align-items)                                         | -                                                                 |
| extStyle      | `string` \| `Record<string, string>`  | 否       | -          | 根节点额外扩展样式                                           | 属性值如果是带单位的话，要带上单位，如：`{margin: '10px'}`       |
| divider       | `boolean` | 否       | -          | 是否展示分割线                                               | -                                                                 |
| titleWidth    | `string`  | 否       | -          | 标题宽度，须包含单位，例如：`100px`                                         | -                                                                 |

### 插槽 **Slots**

| 名称      | 说明                       | 备注 |
| --------- | -------------------------- | ---- |
| icon      | 自定义右侧图标块展示       | 传入 `icon` 属性会忽略此 `slot`   |
| title     | 自定义标题块展示           | 传入 `title` 属性会忽略此 `slot`     |
| label     | 自定义附加信息块展示       |  传入 `label` 属性会忽略标题此 `slot`  |
| desc      | 自定义描述信息块展示       | 传入 `desc` 属性会忽略此 `slot`    |
| subDesc   | 自定义描述信息附加信息展示 | 传入 `subDesc` 属性会忽略此 `slot`    |
| rightIcon | 自定义右侧图标块展示       | 传入 `rightIcon` 或者设置 `arrow` 属性会忽略此 `slot`  |

### 可扩展样式名 **External Class**

| 类名         | 说明                                 | 备注 |
| ------------ | ------------------------------------ | ---- |
| extClass     | 根节点可扩展的类名                   | -    |
| titleClass   | 单元格附加信息的自定义类名           | -    |
| labelClass   | 单元格标题的自定义类名               | -    |
| descClass    | 单元格描述信息的自定义类名           | -    |

### CSS 变量 **CSS Variables**

| 变量                          |  默认值 | 说明 | 备注 |
| --------------------------    |  ------ | ----- | ---- |
| `--cell-padding-v`            | `@gap-32` | - | - |
| `--cell-padding-h`            | `@gap-28` | - | - |
| `--cell-text-color`           | `var(--neutral-color-1, #212121)` | - | - |
| `--cell-title-text-color`     | `var(--neutral-color-1, #212121)` | - | - |
| `--cell-label-text-color`     | `var(--neutral-color-3, #9e9e9e)` | - | - |
| `--cell-desc-text-color`      | `var(--neutral-color-1, #212121)` | - | - |
| `--cell-bg-color`             | `var(--neutral-color-9, #ffffff)` | - | - |
| `--cell-right-icon-color`     | `var(--neutral-color-4, #c4c4c4)` | - | - |
| `--cell-text-disabled-color`  | `var(--neutral-color-3, #9e9e9e)` | - | - |
| `--cell-hover-bg-color`       | `rgba(0, 0, 0, 0.1)` | - | - |
| `--cell-required-color`       | `#ff0000` | - | - |
| `--cell-required-margin-left` | `4px` | - | - |
| `--cell-divider-gap`          | `28px` | - | - |
| `--cell-line-height`          | `1.2` | - | - |
| `--cell-icon-margin-right`    | `20px` | - | - |
| `--cell-title-font-size`      | `28px` | - | - |
| `--cell-title-font-weight`    | `400` | - | - |
| `--cell-title-line-height`    | `42px` | - | - |
| `--cell-label-font-size`      | `24px` | - | - |
| `--cell-label-font-weight`    | `400` | - | - |
| `--cell-label-line-height`    | `36px` | - | - |
| `--cell-desc-font-size`       | `28px` | - | - |
| `--cell-label-font-weight`    | `400` | - | - |
| `--cell-label-line-height`    | `36px` | - | - |
| `--cell-title-wrap-width`    | `auto` | - | - |
| `--cell-title-wrap-min-width`    | `100px` | - | - |

## TiCellGroup API

### 属性 **Properties**

| 名称     | 类型     | 默认值  | 说明               | 备注                                  |
| -------- | -------- | ------- | ------------------ | ------------------------------------- |
| title    | `string` | -       | 标题               | - |
| subTitle | `string` | -       | 副标题             | 只有 `title` 存在时，才会展示此选项。 |
| mode     | `default` \| `card` | `default` | 单元格组展示的模式 | -  |

### 插槽 **Slots**

| 名称    | 说明               | 备注 |
| ------- | ------------------ | ---- |
| title   | 自定义标题块展示   |  传入 `title` 属性会忽略此 `slot`   |

### 可扩展样式名 **External Class**

| 类名       | 说明                   | 备注 |
| ---------- | ---------------------- | ---- |
| extClass   | 根节点可扩展的类名     | -    |
| titleWrapClass | 分组标题的可扩展的类名 | -    |
| titleClass   | 标题可扩展类名       | -    |
| subTitleClass   | 副标题可扩展类名       | -    |

### CSS 变量 **CSS Variables**

| 变量                                  | 默认值                 | 说明 | 备注 |
| ----------------------------------   | -------------------- | ---- | ---- |
| `--cell-group-margin-v`              | `0` | -    | - |
| `--cell-group-margin-h`              | `16px` | -    | - |
| `--cell-group-radius`               | `8px` | - | - |
| `--cell-group-title-padding-v`       | `28px` | - | - |
| `--cell-group-title-padding-h`       | `28px` | - | - |
| `--cell-group-title-color`           | `var(--neutral-color-1, #212121)` | - | - |
| `--cell-group-title-font-size`       | `28px` | - | - |
| `--cell-group-title-font-weight`     | `var(--font-weight-bold, 600)` | - | - |
| `--cell-group-title-line-height`     | `1.2` | - | - |
| `--cell-group-sub-title-color`       | `var(--neutral-color-2, #757575)` | - | - |
| `--cell-group-sub-title-font-size`   | `20px` | - | - |
| `--cell-group-sub-title-font-weight` | `400` | - | - |
| `--cell-group-sub-title-line-height` | `1.2` | - | - |
| `--cell-group-sub-title-margin-left` | `8px` | - | - |
| `--cell-group-body-bg-color`         | `var(--neutral-color-9, #ffffff)` | - | - |
