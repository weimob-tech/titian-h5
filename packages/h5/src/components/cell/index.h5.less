@import '../common/static/var';
@import '../common/static/utils';
@import '../common/static/mixins/index.less';

@cell: ~'@{project}-cell';

:host {
  display: contents;
}

.@{cell} {
  @protected-cell-padding-v: var(--protected-cell-padding-v, @cell-padding-v);
  @protected-cell-padding-h: var(--protected-cell-padding-h, @cell-padding-h);
  @protected-cell-text-color: var(--protected-cell-text-color, @cell-text-color);
  @protected-cell-title-text-color: var(--protected-cell-title-text-color, @cell-title-text-color);
  @protected-cell-label-text-color: var(--protected-cell-label-text-color, @cell-label-text-color);
  @protected-cell-desc-text-color: var(--protected-cell-desc-text-color, @cell-desc-text-color);
  @protected-cell-bg-color: var(--protected-cell-bg-color, @cell-bg-color);
  @protected-cell-right-icon-color: var(--protected-cell-right-icon-color, @cell-right-icon-color);
  @protected-cell-text-disabled-color: var(--protected-cell-text-disabled-color, @cell-text-disabled-color);
  @protected-cell-hover-bg-color: var(--protected-cell-hover-bg-color, @cell-hover-bg-color);
  @protected-cell-required-color: var(--protected-cell-required-color, @cell-required-color);

  display: flex;
  position: relative;
  box-sizing: border-box;
  justify-content: space-between;
  width: 100%;
  padding: @protected-cell-padding-v @protected-cell-padding-h;

  transition-property: background-color;
  transition-duration: 0.1s;
  transition-timing-function: linear;

  background-color: @protected-cell-bg-color;

  color: @protected-cell-text-color;
  line-height: @cell-line-height;
  word-break: @cell-word-break;

  --icon-line-height: 1;

  &-divider::after {
    .hairline(bottom, @neutral-color-6);

    right: @gap-28;
    left: @gap-28;
  }

  &-disabled {
    pointer-events: none;

    .@{cell}-title,
    .@{cell}-label,
    .@{cell}-desc,
    .@{cell}-right-icon {
      color: @protected-cell-text-disabled-color;
    }
  }

  &-clickable:active {
    background-color: @protected-cell-hover-bg-color;
  }

  &-icon-box,
  &-required-box {
    display: flex;
    align-items: center;
    min-height: 42px;
  }

  &-icon {
    margin-right: @cell-icon-margin-right;
  }

  &-title {
    color: @protected-cell-title-text-color;
    font-size: @cell-title-font-size;
    font-weight: @cell-title-font-weight;
    line-height: @cell-title-line-height;
  }

  &-label {
    color: @protected-cell-label-text-color;
    font-size: @cell-label-font-size;
    font-weight: @cell-label-font-weight;
    line-height: @cell-label-line-height;
  }

  &-required {
    margin-left: @cell-required-margin-left;

    --protected-icon-line-height: @T42;
    --icon-color: @protected-cell-required-color;
  }

  &-title-wrap,
  &-desc-wrap {
    display: flex;
    justify-content: center;
  }

  &-title-wrap {
    align-items: center;
    justify-content: flex-start;
    width: @cell-title-wrap-width;
    min-width: @cell-title-wrap-min-width;
  }

  &-desc-wrap {
    flex-direction: column;
    align-items: flex-end;
  }

  &-desc-content {
    display: inline-flex;
    align-items: center;
    min-height: 42px;
  }

  &-desc {
    color: @protected-cell-desc-text-color;
    font-size: @cell-desc-font-size;
    font-weight: @cell-desc-font-weight;
    line-height: @cell-desc-line-height;
  }

  &-sub-desc {
    box-sizing: border-box;
    width: 100%;

    color: @protected-cell-label-text-color;
    font-size: @T24;
    font-weight: @font-weight-400;
    line-height: @T36;
    text-align: right;
  }

  &-right-icon {
    margin-left: @cell-right-icon-margin-left;

    --icon-color: @protected-cell-right-icon-color;
  }
}
