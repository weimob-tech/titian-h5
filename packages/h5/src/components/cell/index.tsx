import { Component, Element, h, Prop } from '@stencil/core';
import { JSXBase } from '@stencil/core/internal';
import { BasicComponentAbstract } from '../common/basic/BasicComponent';
import { isString, stringToAttrStyle, addShadowRootStyle } from '../common/utils';
import { join, handle } from '../common/utils/namespace';
import addUnit from '../common/utils/suffix';

@Component({
  tag: 'ti-cell',
  styleUrls: {
    pc: 'index.pc.less',
    h5: 'index.h5.less',
  },
  shadow: true,
})
export class TiCell implements BasicComponentAbstract {
  @Prop() extCss = '';

  @Prop() extClass?: string;

  @Prop() titleWidth?: string;

  @Prop() extStyle?: string | JSXBase.HTMLAttributes<Record<string, unknown>>['style'];

  /**
   * 标题
   *
   * @type string
   * @default ''
   * @example
   * <TiCell title="标题" />
   * @since 0.1.0
   * */
  @Prop() title: string | any;

  /**
   * 副标题，右侧描述信息
   *
   * @type string
   * @default ''
   * @example
   * <TiCell title="标题" desc="副标题" />
   * @since 0.1.0
   * */
  @Prop() desc?: string;

  /**
   * 标题下面的内容
   *
   * @type string
   * @default ''
   * @example
   * <TiCell title="标题" label="标题下面的内容" />
   * @since 0.1.0
   * */
  @Prop() label?: string;

  /**
   * 副标题下面的内容
   *
   * @type string
   * @default ''
   * @example
   * <TiCell subDesc="副标题下面的内容" />
   * @since 0.1.0
   * */
  @Prop() subDesc?: string;

  /**
   * 图标
   *
   * @type string
   * @default ''
   * @example
   * <TiCell title="标题" icon="home" />
   * @since 0.1.0
   * */
  @Prop() icon?: string;

  /**
   * 右侧图标的尺寸
   *
   * @type number
   * @default 36
   * @example
   * <TiCell iconSize={36} />
   * @since 0.1.0
   */
  @Prop() iconSize?: string | number = 36;

  /**
   * 右侧图标
   *
   * @type string
   * @default ''
   * @example
   * <TiCell title="标题" rightIcon="checked" />
   * @since 0.1.0
   * */
  @Prop() rightIcon?: string;

  /**
   * 右侧图标的尺寸
   *
   * @type number
   * @default 28
   * @example
   * <TiCell right-icon-size="{{ 28 }}" />
   * @since 0.1.0
   */
  @Prop() rightIconSize?: string | number = 28;

  /**
   * 图标颜色
   *
   * @type string
   * @default ''
   * @example
   * <TiCell color="red" />
   * @since 0.1.0
   *
   */
  @Prop() color?: string;

  /**
   * 是否显示右侧箭头
   *
   * @type boolean
   * @default true
   * @example
   * <TiCell arrow="{{false}}" />
   * @since 0.1.0
   * */
  @Prop() arrow?: boolean = false;

  /**
   * 是否禁用单元格
   *
   * @type boolean
   * @default false
   * @example
   * <TiCell title="标题" disabled />
   * @since 0.1.0
   * */
  @Prop() disabled?: boolean = false;

  /**
   * 是否必须选择
   *
   * @type boolean
   * @default false
   * @example
   * <TiCell required />
   * @since 0.1.0
   * */
  @Prop() required?: boolean = false;

  /**
   * 是否显示分割线
   *
   * @type boolean
   * @default true
   * @example
   * <TiCell divider={false} title="没有分割线" />
   * @since 0.1.0
   */
  @Prop() divider?: boolean = true;

  /**
   * 是否开启点击反馈
   *
   * @type boolean
   * @default true
   * @example
   * <TiCell clickable />
   * @since 0.1.0
   * */
  @Prop() clickable?: boolean = false;

  /**
   * 内容块的横向排列方式
   *
   * @type string
   * @default center
   * @example
   * <TiCell alignItems="center" />
   * @since 0.1.0
   * */
  @Prop() alignItems: 'center' | 'flex-start' | 'flex-end' | 'baseline' | 'stretch' | 'start' | 'end';

  @Prop() titleClass?: string;

  @Prop() titleWrapClass?: string;

  @Prop() labelClass?: string;

  @Prop() descClass?: string;

  @Prop({ reflect: true }) useSubArrow?: boolean = false;

  /**
   * 点击单元格回调的方法
   *
   * @param event TouchEvent
   * @return
   *
   * @example
   * <TiCell onClick="onClick" />
   * @since 0.1.0
   * */
  // @Event({ bubbles: true, composed: true }) click: EventEmitter<Record<string, any>>;

  @Element() protected host: HTMLDivElement;

  completedStyles(extStyle = {}) {
    return { ...stringToAttrStyle(extStyle), alignItems: this.alignItems };
  }

  private getClassName(): string {
    const { disabled, divider, extClass, clickable } = this;
    const classList = [join('cell', { disabled, divider, clickable }), extClass];

    return classList.filter(Boolean).join(' ');
  }

  private completedArrow(): string {
    if (isString(this.rightIcon) && this.rightIcon) {
      return this.rightIcon;
    }
    return 'arrow-right';
  }

  private completedRightIcon() {
    const { desc, rightIcon, rightIconSize, arrow } = this;
    if (!desc) {
      return { 'display': 'flex', 'justify-content': 'flex-end' };
    }
    if (rightIcon || arrow) {
      return { paddingRight: addUnit(+rightIconSize + 4) };
    }
    return { 'display': 'flex', 'justify-content': 'flex-end' };
  }

  renderLeftIcon() {
    return (
      <div class={handle('cell', 'icon-box')}>
        {this.icon ? (
          <ti-icon name={this.icon} size={this.iconSize || 36} color={this.color} class={handle('cell', 'icon')} />
        ) : (
          <slot name="icon" />
        )}
      </div>
    );
  }

  renderArrowIcon() {
    if (this.useSubArrow) {
      return null;
    }
    if (this.rightIcon || this.arrow) {
      return (
        <ti-icon
          color={this.color}
          size={this.rightIconSize}
          name={this.completedArrow()}
          ext-class={handle('cell', 'right-icon')}
        />
      );
    }
    return <slot name="right-icon" />;
  }

  componentWillLoad() {
    addShadowRootStyle.call(this);
  }

  render() {
    return (
      <div
        aria-hidden="true"
        part={this.extClass}
        class={this.getClassName()}
        style={this.completedStyles(this.extStyle)}
      >
        <div
          part={this.titleWrapClass}
          class={`${handle('cell', 'title-wrap')} ${this.titleWrapClass || ''}`}
          aria-hidden="true"
          style={this.completedStyles()}
        >
          {this.renderLeftIcon()}
          <div>
            <div style={{ ...this.completedStyles(), display: 'flex' }}>
              {this.title ? (
                <div
                  part={this.titleClass}
                  class={`${handle('cell', 'title')} ${this.titleClass}`}
                  style={this.titleWidth ? { width: this.titleWidth } : {}}
                >
                  {this.title ?? null}
                </div>
              ) : (
                <slot name="title" />
              )}
              {this.required ? (
                <div class={handle('cell', 'required-box')}>
                  <ti-icon name="required" size={24} ext-class={handle('cell', 'required')} />
                </div>
              ) : null}
            </div>

            {this.label ? (
              <div part={this.labelClass} class={`${handle('cell', 'label')} ${this.labelClass}`}>
                {this.label ?? null}
              </div>
            ) : (
              <slot name="label" />
            )}
          </div>
        </div>
        <div class={handle('cell', 'desc-wrap')} style={{ alignItems: this.alignItems }}>
          <div class={handle('cell', 'desc-content')} aria-hidden="true">
            {this.desc ? (
              <div part={this.descClass} class={`${handle('cell', 'desc')} ${this.descClass}`}>
                {this.desc ?? null}
              </div>
            ) : (
              <slot name="desc" />
            )}
            {this.renderArrowIcon()}
          </div>
          {this.subDesc ? (
            <div style={this.completedRightIcon()} class={handle('cell', 'sub-desc')}>
              {this.subDesc ?? null}
            </div>
          ) : (
            <slot name="sub-desc" />
          )}
        </div>
      </div>
    );
  }
}
