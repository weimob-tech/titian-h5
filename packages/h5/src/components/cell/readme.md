# ti-cell

<!-- Auto Generated Below -->


## Properties

| Property        | Attribute         | Description      | Type                                                                | Default     |
| --------------- | ----------------- | ---------------- | ------------------------------------------------------------------- | ----------- |
| `alignItems`    | `align-items`     | 内容块的横向排列方式       | `"baseline" \| "center" \| "flex-end" \| "flex-start" \| "stretch"` | `undefined` |
| `arrow`         | `arrow`           | 是否显示右侧箭头         | `boolean`                                                           | `false`     |
| `clickable`     | `clickable`       | 是否开启点击反馈         | `boolean`                                                           | `true`      |
| `color`         | `color`           | 图标颜色             | `string`                                                            | `undefined` |
| `desc`          | `desc`            | 副标题，右侧描述信息       | `string`                                                            | `undefined` |
| `descClass`     | `desc-class`      |                  | `string`                                                            | `undefined` |
| `disabled`      | `disabled`        | 是否禁用单元格          | `boolean`                                                           | `false`     |
| `divider`       | `divider`         | 是否显示分割线          | `boolean`                                                           | `true`      |
| `extClass`      | `ext-class`       | 额外的类名，添加到根节点的元素上 | `string`                                                            | `undefined` |
| `extStyle`      | --                | 额外的样式            | `{}`                                                                | `{}`        |
| `icon`          | `icon`            | 图标               | `string`                                                            | `undefined` |
| `iconSize`      | `icon-size`       | 右侧图标的尺寸          | `number \| string`                                                  | `36`        |
| `label`         | `label`           | 标题下面的内容          | `string`                                                            | `undefined` |
| `labelClass`    | `label-class`     |                  | `string`                                                            | `undefined` |
| `required`      | `required`        | 是否必须选择           | `boolean`                                                           | `false`     |
| `rightIcon`     | `right-icon`      | 右侧图标             | `string`                                                            | `undefined` |
| `rightIconSize` | `right-icon-size` | 右侧图标的尺寸          | `number \| string`                                                  | `28`        |
| `subDesc`       | `sub-desc`        | 副标题下面的内容         | `string`                                                            | `undefined` |
| `title`         | `title`           | 标题               | `any`                                                               | `undefined` |
| `titleClass`    | `title-class`     |                  | `string`                                                            | `undefined` |
| `useSubArrow`   | `use-sub-arrow`   |                  | `boolean`                                                           | `false`     |


## Events

| Event   | Description | Type                                 |
| ------- | ----------- | ------------------------------------ |
| `click` | 点击单元格回调的方法  | `CustomEvent<{ [x: string]: any; }>` |


## Dependencies

### Used by

 - [ti-collapse-item](../collapse-item)
 - [ti-dropdown-item](../dropdown-item)

### Depends on

- [ti-icon](../icon)

### Graph
```mermaid
graph TD;
  ti-cell --> ti-icon
  ti-collapse-item --> ti-cell
  ti-dropdown-item --> ti-cell
  style ti-cell fill:#f9f,stroke:#333,stroke-width:4px
```

----------------------------------------------

*Built with [StencilJS](https://stenciljs.com/)*
