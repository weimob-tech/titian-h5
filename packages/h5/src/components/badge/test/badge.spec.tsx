import { newSpecPage } from '@stencil/core/testing';
import { h } from '@stencil/core';
import { TiBadge } from '../index';
import { TiCell } from '../../cell';
import { TestContainer } from '../../common/test/container';

describe('渲染 ti-action-sheet', () => {
  it('基本渲染', async () => {
    const page = await newSpecPage({
      components: [TiBadge, TestContainer],
      template: () => (
        <test-container>
          <ti-badge content="5">
            <div class="box" />
          </ti-badge>
          <ti-badge dot>
            <div class="box" />
          </ti-badge>
          <ti-badge icon="home">
            <div class="box" />
          </ti-badge>
          <div class="box" style={{ position: 'relative' }}>
            <ti-badge content="6" />
          </div>
        </test-container>
      ),
    });
    expect(page.root).toMatchSnapshot();
  });

  it('无定位', async () => {
    const page = await newSpecPage({
      components: [TiBadge, TiCell],
      template: () => (
        <ti-cell>
          <ti-badge static slot="desc" content="6" />
        </ti-cell>
      ),
    });
    expect(page.root).toMatchSnapshot();
  });

  it('自定义偏移', async () => {
    const page = await newSpecPage({
      components: [TiBadge],
      template: () => (
        <ti-badge content="6" offset={[2, 6]}>
          <div class="box" />
        </ti-badge>
      ),
    });
    expect(page.root).toMatchSnapshot();
  });

  it('在文字右上角', async () => {
    const page = await newSpecPage({
      components: [TiBadge],
      template: () => (
        <ti-badge content="6" at-text>
          <div>Title Text</div>
        </ti-badge>
      ),
    });
    expect(page.root).toMatchSnapshot();
  });

  it('徽标内容撑开方向', async () => {
    const page = await newSpecPage({
      components: [TiBadge, TestContainer],
      template: () => (
        <test-container>
          <ti-badge content="6" spread="bothSides">
            <div class="box" />
          </ti-badge>
          <ti-badge content="6" spread="toRight">
            <div class="box" />
          </ti-badge>
        </test-container>
      ),
    });
    expect(page.root).toMatchSnapshot();
  });
});
