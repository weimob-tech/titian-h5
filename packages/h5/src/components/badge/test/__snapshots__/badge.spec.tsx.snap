// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`渲染 ti-action-sheet 在文字右上角 1`] = `
<ti-badge at-text="">
  <mock:shadow-root>
    <div aria-hidden="true" class="titian-badge">
      <div class="titian-badge-container titian-badge-container-atText titian-badge-container-circle" part="">
        6
      </div>
      <slot></slot>
    </div>
  </mock:shadow-root>
  <div>
    Title Text
  </div>
</ti-badge>
`;

exports[`渲染 ti-action-sheet 基本渲染 1`] = `
<test-container>
  <!---->
  <ti-badge>
    <mock:shadow-root>
      <div aria-hidden="true" class="titian-badge">
        <div class="titian-badge-container titian-badge-container-circle" part="">
          5
        </div>
        <slot></slot>
      </div>
    </mock:shadow-root>
    <div class="box"></div>
  </ti-badge>
  <ti-badge>
    <mock:shadow-root>
      <div aria-hidden="true" class="titian-badge">
        <div class="titian-badge-container titian-badge-container-dot" part=""></div>
        <slot></slot>
      </div>
    </mock:shadow-root>
    <div class="box"></div>
  </ti-badge>
  <ti-badge>
    <mock:shadow-root>
      <div aria-hidden="true" class="titian-badge">
        <div class="titian-badge-container titian-badge-container-circle" part="">
          <ti-icon name="home" size="20"></ti-icon>
        </div>
        <slot></slot>
      </div>
    </mock:shadow-root>
    <div class="box"></div>
  </ti-badge>
  <div class="box" style="position: relative;">
    <ti-badge>
      <mock:shadow-root>
        <div aria-hidden="true" class="titian-badge">
          <div class="titian-badge-container titian-badge-container-circle" part="">
            6
          </div>
          <slot></slot>
        </div>
      </mock:shadow-root>
    </ti-badge>
  </div>
</test-container>
`;

exports[`渲染 ti-action-sheet 徽标内容撑开方向 1`] = `
<test-container>
  <!---->
  <ti-badge>
    <mock:shadow-root>
      <div aria-hidden="true" class="titian-badge">
        <div class="titian-badge-container titian-badge-container-circle" part="">
          6
        </div>
        <slot></slot>
      </div>
    </mock:shadow-root>
    <div class="box"></div>
  </ti-badge>
  <ti-badge>
    <mock:shadow-root>
      <div aria-hidden="true" class="titian-badge">
        <div class="titian-badge-container titian-badge-container-circle titian-badge-container-left" part="">
          6
        </div>
        <slot></slot>
      </div>
    </mock:shadow-root>
    <div class="box"></div>
  </ti-badge>
</test-container>
`;

exports[`渲染 ti-action-sheet 无定位 1`] = `
<ti-cell>
  <mock:shadow-root>
    <div aria-hidden="true" class="titian-cell titian-cell-divider">
      <div aria-hidden="true" class="titian-cell-title-wrap">
        <div class="titian-cell-icon-box">
          <slot name="icon"></slot>
        </div>
        <div>
          <div style="display: flex;">
            <slot name="title"></slot>
          </div>
          <slot name="label"></slot>
        </div>
      </div>
      <div class="titian-cell-desc-wrap">
        <div aria-hidden="true" class="titian-cell-desc-content">
          <slot name="desc"></slot>
          <slot name="right-icon"></slot>
        </div>
        <slot name="sub-desc"></slot>
      </div>
    </div>
  </mock:shadow-root>
  <ti-badge slot="desc">
    <mock:shadow-root>
      <div aria-hidden="true" class="titian-badge">
        <div class="titian-badge-container titian-badge-container-circle titian-badge-container-static" part="">
          6
        </div>
        <slot></slot>
      </div>
    </mock:shadow-root>
  </ti-badge>
</ti-cell>
`;

exports[`渲染 ti-action-sheet 自定义偏移 1`] = `
<ti-badge>
  <mock:shadow-root>
    <div aria-hidden="true" class="titian-badge">
      <div class="titian-badge-container titian-badge-container-circle titian-badge-container-offset" part="" style="--badge-top: 1px; --badge-right: 3px;">
        6
      </div>
      <slot></slot>
    </div>
  </mock:shadow-root>
  <div class="box"></div>
</ti-badge>
`;
